sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -103,6 +107,75 @@
 # Helper functions =============================================================
 #
 
+
+def is_cmis_api(api):
+   return type(api) == CmisApi
+
+
+def get_cmis_application_desired(api, host_lane_count, speed):
+    """
+    Get the CMIS application code that matches the specified host side configurations
+
+    Args:
+        api:
+            XcvrApi object
+        host_lane_count:
+            Number of lanes on the host side
+        speed:
+            Integer, the port speed of the host interface
+
+    Returns:
+        Integer, the transceiver-specific application code
+    """
+
+    if speed == 0 or host_lane_count == 0:
+        return None
+
+    if not is_cmis_api(api):
+        return None
```

Comment:
@tshalvi the caller for this is CMIS task manager which calls this API only for CMIS compliant transceivers. So we don't need this check again

Reply:
The is_cmis_api() method is called not only within the CmisManagerTask, but is also invoked from the get_lane_speed_key() method in the media_settings_parser during the notify_media_setings() process (within SfpStateUpdateTask). When it's called from this method, there's no guarantee that the module being used is a CMIS module.